File: 05_Recursion/Indirect_Recursion /main.go
package main

import "fmt"

//12 11 8 7 4 3

// func fun1(a int) {
// 	if a > 0 {
// 		fmt.Printf("%d ", a)
// 		fun2(a - 1)
// 	}
// }

// func fun2(b int) {
// 	if b > 0 {
// 		fmt.Printf("%d ", b)
// 		fun1(b - 3)
// 	}
// }
// func main() {
// 	v := 12
// 	fun1(v)
// }

//3 4 7 8 11 12

func fun1(a int) {
	if a > 0 {
		fun2(a - 1)
		fmt.Printf("%d ", a)
	}
}

func fun2(b int) {
	if b > 0 {
		fun1(b - 3)
		fmt.Printf("%d ", b)
	}
}
func main() {
	v := 12
	fun1(v)
}

File: 05_Recursion/Indirect_Recursion /main.cpp
// #include <stdio.h>
// void fun2(int b);

// void fun1(int a){
//     if (a > 0)
//     {
//         printf("%d\n", a);
//         fun2(a - 1);
//     }
// }

// void fun2(int b){
//     if(b > 0)
//     {
//         printf("%d\n", b);
//         fun1(b - 3);
//     }
// }

// int main (){
//     int v = 12; 
//     fun1 (v);
//     return 0;
// }

//3 4 7 8 11 12

#include <stdio.h>
void fun2(int b);

void fun1(int a){
    if (a > 0)
    {
        fun2(a - 1);
        printf("%d ", a);
    }
}

void fun2(int b){
    if(b > 0)
    {
        fun1(b - 3);
        printf("%d ", b);
    }
}

int main (){
    int v = 12; 
    fun1 (v);
    return 0;
}

